open-metadata-resources/open-metadata-deployment/charts/vdc/templates/job-01-data-loader.yaml

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Contributors to the Egeria project.
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "myapp.fullname" . }}-job-01-data-loader
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    helm.sh/chart: {{ include "myapp.chart" . }}
+    app.kubernetes.io/name: {{ include "myapp.name" . }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/name: {{ include "myapp.name" . }}
+    spec:
+      restartPolicy: OnFailure
+      volumes:
+        - name: samples-volume
+          emptyDir: {}
+      initContainers:
+        - name: init-git
+          image: "{{ .Values.image.configure.repository }}:{{ .Values.image.configure.tag }}"
+          command:
+            - "/bin/bash"
+            - "-c"
+            - >
+                cd /samples &&
+                git init egeria &&
+                cd egeria &&
+                git remote add origin https://github.com/odpi/egeria.git &&
+                git config core.sparsecheckout true &&
+                echo 'open-metadata-resources/open-metadata-deployment/sample-data/*' >> .git/info/sparse-checkout &&
+                git pull --depth=1 origin master
+          volumeMounts:
+            - mountPath: /samples
+              name: samples-volume
+        - name: wait-for-db
+          image: busybox
+          command:
+            - "sh"
+            - "-c"
+            - >
+                until nslookup {{ include "myapp.fullname" . }}-postgresql-service; do
+                echo waiting for {{ include "myapp.fullname" . }}-postgresql-service;
+                sleep 2;
+                done;
```

Comment:
Will this be sufficient? JUst the name being available? Perhaps using 'netcat' to check the port can be connected to is one step better? (obviously valid postgresql request even better). I found netcat generally good enough in the initial scripts

Reply:
Good catch -- I'd replaced most of this with a more robust check (basically introspecting k8s itself to see that there is at least one pod available and ready behind the service) before continuing -- just pushed a change for this one as well.